MSK SASL/SCRAM using Config Providers

nicusX opened the pull request:
Example of configuring Kafka connector for SASL/SCRAM fetching secrets and truststore at runtime, using Kafka Config Providers.

jeremyber-aws reviewed (approved):
Code looks well documented and makes sense

nicusX commented:
> just thinking for sake of simplicity does it make sense to put this under same folder as other kafka config providers example? maybe rename top level directory?

Moved both examples using config providers under a directory, and renamed the example directories for consistency
